core/constraints/constraints.go

```diff
@@ -39,6 +39,18 @@ const (
 	ImageID          = "image-id"
 )
 
+// CharmValue describes a user's requirements of the hardware on which units
+// of an application will run. Constraints are used to choose an existing machine
+// onto which a unit will be deployed, or to provision a new machine if no
+// existing one satisfies the requirements.
```

Comment:
This comment isn't correct.
Left move the struct to the caas package as it's a k8s specific artefact.